pkg/reconciler/workload/synctarget/synctarget_controller.go

```diff
@@ -42,21 +42,29 @@ import (
 	kcpclientset "github.com/kcp-dev/kcp/sdk/client/clientset/versioned/cluster"
 	corev1alpha1informers "github.com/kcp-dev/kcp/sdk/client/informers/externalversions/core/v1alpha1"
 	workloadv1alpha1informers "github.com/kcp-dev/kcp/sdk/client/informers/externalversions/workload/v1alpha1"
-	corev1alpha1listers "github.com/kcp-dev/kcp/sdk/client/listers/core/v1alpha1"
 )
 
 const ControllerName = "kcp-synctarget-controller"
 
 func NewController(
 	kcpClusterClient kcpclientset.ClusterInterface,
 	syncTargetInformer workloadv1alpha1informers.SyncTargetClusterInformer,
-	workspaceShardInformer corev1alpha1informers.ShardClusterInformer,
+	workspaceShardInformer, globalWorkspaceShardInformer corev1alpha1informers.ShardClusterInformer,
 ) *Controller {
 	c := &Controller{
-		queue:                workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName),
-		kcpClusterClient:     kcpClusterClient,
-		syncTargetIndexer:    syncTargetInformer.Informer().GetIndexer(),
-		workspaceShardLister: workspaceShardInformer.Lister(),
+		queue:             workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName),
+		kcpClusterClient:  kcpClusterClient,
+		syncTargetIndexer: syncTargetInformer.Informer().GetIndexer(),
+		listWorkspaceShards: func(selector labels.Selector) ([]*corev1alpha1.Shard, error) {
+			shards, err := workspaceShardInformer.Lister().List(selector)
+			if len(shards) == 0 {
+				return globalWorkspaceShardInformer.Lister().List(selector)
+			}
+			if err != nil {
+				return nil, err
+			}
+			return shards, nil
```

Comment:
Here only the local list is returned. But I assume it is OK for `WorkspaceShard`s since all the `WorkspaceShard` objects are living on the same shard (the shard of the `root` workspace). Is it the reason why this is done this way ?